opm/simulators/linalg/setupPropertyTree_impl.hpp

```diff
@@ -39,7 +39,7 @@ setupPropertyTree(FlowLinearSolverParameters p) // Note: copying the parameters
     std::string conf = p.linsolver_;
 
     // Get configuration from file.
-    if (conf.substr(conf.size() - 5, 5) == ".json") {
+    if (conf.size() > 5 && conf.substr(conf.size() - 5, 5) == ".json") { // the ends_with() method is not available until C++20
```

Comment:
Just out of curiosity, did you consider using `string::rfind()`?  E.g., something like
```C++
if (conf.rfind(".json") != std::string::npos) {
    // 'conf' is *.json
}
```

Reply:
Putting my smart glasses on and playing smart-ass:

Doesn't really cut it as it would return true for file finame.json.txt, too.

Reply:
> Just out of curiosity, did you consider using `string::rfind()`? E.g., something like

I did not. I see that would be a little easier, but it does not do exactly the same: it would find the string anywhere, not just at the end. I guess that is mostly academic though.